mtz-update-password.html

```diff
@@ -303,6 +417,49 @@
           computed: `__buildValidationValues(allowedCharacters, allowedSymbols, maxlength, minlength, minUpper,
             minLetters, minLower, minNumbers, minSymbols)`,
         },
+        __passwordStrengthMeterValue: String,
+        __passwordStrengthTextValue: {
+          type: String,
+          value: ''
+        },
+        __checked: {
+          type: Boolean,
+          value: false,
+          observer: '__setPasswordType'
+        }
+      },
+      observers: [
+        '__updateStrengthMeter(value)'
+      ],
+      /**
+       * Update strength meter indicator and strength text
+       * @private
+       *
+       * @param {String} value
+       */
+      __updateStrengthMeter(value) {
+
+        //do not run strength meter when its hidden
+        if (!this.showStrengthMeter) return;
+
+        this.importLazyGroup('zxcvbn').then(() => {
+          const result = zxcvbn(value);
+          // Update the password strength meter
+          this.__passwordStrengthMeterValue = result.score;
+
+          // Update the text indicator
+          this.__passwordStrengthTextValue = value !== '' ?
+            `Strength:&nbsp;<span class="result-color${result.score}">${this.strength[result.score]}</span>` : '';
+        });
+      },
+      /**
+       * set password type when the reveal icon is clicked
+       * @private
+       *
+       * @param {Boolean} checked
+       */
+      __setPasswordType(checked) {
+        this.$.password.type = checked ? 'text' : 'password';
```

Comment:
Instead of using DOM manipulation directly, change this to `return checked ? 'text' : 'password';` and update the template above to use `type$="[[__setPasswordType(__checked)]]"`, you can then drop the property observer.